c7n/resources/ec2.py

```diff
@@ -653,6 +653,82 @@ def process_volume_set(self, resource):
                 Tags=tags)
 
 
+@actions.register('rename-tag')
+class RenameTag(BaseAction):
+
+    schema = type_schema(
+        'rename-tag',
+        map={'type': 'list', 'items': {'type': 'string'}})
+
+    def delete_tag(self, client, ec2, key, value):
+        client.delete_tags(
+            Resources=[ec2['InstanceId']],
+            Tags=[{'Key': key, 'Value': value}])
+
+    def create_tag(self, client, ec2, key, value):
+        client.create_tags(
+            Resources=[ec2['InstanceId']],
+            Tags=[{'Key': key, 'Value': value}])
+
+    def process_ec2(self, source_tag, source_value, resources):
+        """
+        Move source tag value to destination tag value
+
+        - Collect value from old tag
+        - Delete old tag
+        - Create new tag & assign stored value
+        """
+        tag_map = self.data.get('map', [{}])[0]
+        dest_tag = tag_map[source_tag]
+        over = []
+        under = []
+        for r in resources:
+            if len(r.get('Tags', [])) < 50:
+                under.append(r['InstanceId'])
+            else:
+                over.append(r['InstanceId'])
+
+        c = utils.local_session(self.manager.session_factory).client('ec2')
+        if under:
+            self.create_tag(c, under, dest_tag, source_value)
+
+        self.delete_tag(
+            c, [r['InstanceId'] for r in resources], source_tag, source_value)
+        
+        if over:
+            self.create_tag(c, over, dest_tag, source_value)
+
+    def create_map(self, resources):
+        tag_map = self.data.get('map', [{}])[0]
+        keys = {}
+        for t in tag_map.keys():
```

Comment:
this logic feels convoluted. your iterating over one map to populate another, you'll never hit a case where its in the one being populated.